transcendence-app/src/chat/chat/infrastructure/db/postgres/chat-message.postgres.repository.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+
+import { table } from '../../../../../shared/db/models';
+import { PostgresPool } from '../../../../../shared/db/postgres/postgresConnection.provider';
+import { BasePostgresRepository } from '../../../../../shared/db/postgres/postgres.repository';
+import { ChatMessage, chatMessageKeys } from '../chat-message.entity';
+import { IChatMessageRepository } from '../chat-message.repository';
+import { makeQuery } from '../../../../../shared/db/postgres/utils';
+import { PaginationQueryDto } from '../../../../../shared/dtos/pagination.query.dto';
+
+@Injectable()
+export class ChatMessagePostgresRepository
+  extends BasePostgresRepository<ChatMessage>
+  implements IChatMessageRepository
+{
+  constructor(protected pool: PostgresPool) {
+    super(pool, table.CHAT_MESSAGE, ChatMessage);
+  }
+  async getPaginatedMessages(
+    userMeId: string,
+    recipientId: string,
+    paginationQueryDto: Required<PaginationQueryDto>,
+  ): Promise<ChatMessage[] | null> {
+    const { limit, offset } = paginationQueryDto;
+
+    return makeQuery<ChatMessage>(this.pool, {
```

Comment:
suggestion: Please update this method to return an array of `ChatMessage` instances.

Reply:
I think this is boilerplate since we are not serializing anything. That being said I've added to be consistent with other repositories

Reply:
Thanks